File: layout/macros/examples/function-shape.rb
require 'LayoutScript'
include LayoutScript

# create a polygon on the active layer with a entered function

def func( d)
        return d*d/100
end

l=Project.new_layout()

dr=l.drawing # pointer to the main drawing

dr.currentCell.cellName="ruby_example_function"

dr.clear_points()
dr.activeLayer=4

# add a x^2 polygon
for x in (-100..100).step(10)
        dr.p(x,func(x))
end

dr.polygon()

#add a cos path

dr.activeLayer=2

for x in (-130..130).step(5)
        dr.p(x,-30*Math.cos((x.to_f-20)/50*Math::PI))
end 
dr.path() 


# save to your home folder
l.drawing.save_file("#{Dir.home}/testout.gds")

puts "Ruby script completed" 

File: layout/macros/examples/function-shape.py
import LayoutScript
from LayoutScript import *

# create a polygon on the active layer with a entered function

def func( d):
        return d*d/100
        

l=project.newLayout(); 

dr=l.drawing # pointer to the main drawing

dr.currentCell.cellName="python_example_function"

dr.clearPoints()
dr.activeLayer=4

# add a x^2 polygon
for x in range (-100,101,10):
        dr.p(x,func(x))

dr.polygon()

#add a cos path
import math
dr.activeLayer=2

for x in range (-130,130,5):
        dr.p(x,-30*math.cos((float(x)-20)/50*math.pi))
       
dr.path() 

# writes result to the home folder

import os
l.drawing.saveFile( os.path.expanduser('~')+"/testout.gds")


print ("Python script completed" )


